src/lib/yacc/grammar.rs

```diff
@@ -684,4 +860,69 @@ mod test {
         assert!(!grm.has_path(b_nt_idx, a_nt_idx));
         assert!(!grm.has_path(c_nt_idx, a_nt_idx));
     }
+
+    #[test]
+    fn test_nonterm_costs() {
+        let grm = yacc_grm(YaccKind::Original, &"
```

Comment:
This test could be slightly improved by having a rule `E: 'x' A | 'x' 'y'` to check that the algorithm tests `'x' A` which is shorter than `xy`.